### ISSUE TITLE
bug in examples, not all dimensions listed?

### ISSUE THREAD

**Issue by bklaas** [Dirguis/ipfn#3]
Hi Marius,

I'm wondering if you could help me understand something I'm seeing in the examples you've included with the ipfn code. When going beyond 2 dimensions, I see that the cross dimensions are not fully recursive. That is, dma x size and size x age are present below, but not dma x age. 

    ipfn_df = ipfn(df, [xipp, xpjp, xppk, xijp, xpjk],
               [['dma'], ['size'], ['age'], ['dma', 'size'], ['size', 'age']])

This is true in all of the > 2D examples. My question is, is that an oversight or am I missing something with regards to how to define the aggregates and dimensions?

I appreciate any insight you may have. 

Cheers,
Ben

**Comment by Dirguis** [comment]
Hello Ben

So yes, it all depends on the information that you have. In my case, I pretended that I did not have the marginals for the combination dma+age. Not having that information should not prevent you from running the IPF. After all, you still have the dma+size and size+age marginal information and can still get weights using that information. It would be more complete  if you had dma+age, but it is not a requirement and in many situations, one may not have access to all the possible marginals from all the possible features combinations.

I hope that makes sense. Let me know if something is not clear!

Cheers

Damien
